FileIO/OpenGeoSysProject.xsd

```diff
@@ -2,57 +2,183 @@
+        <xs:complexType>
+          <xs:sequence>
+            <xs:element name="source_term" type="SourceTermType" minOccurs="0" maxOccurs="unbounded" />
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="NonlinearSolverType">
+    <xs:sequence>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="tol" type="xs:double" minOccurs="1" maxOccurs="1" />
+      <xs:element name="max_iter" type="xs:nonNegativeInteger" minOccurs="1" maxOccurs="1" />
+      <xs:element name="linear_solver" type="xs:string" minOccurs="1" maxOccurs="1" />
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="LinearSolverType">
+    <xs:sequence>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <!-- undefined content -->
+      <xs:any processContents="lax" maxOccurs="unbounded"/>
+    </xs:sequence>
   </xs:complexType>
-  
-  
-  <!-- definition of file content -->    
+
+  <!-- definition of file content -->
   <xs:element name="OpenGeoSysProject">
     <xs:complexType>
       <xs:sequence>
-        <xs:element name="geo" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-        <xs:element name="msh" type="mshElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="cnd" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="mfp" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="mmp" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="msp" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="np"  type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="out" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="pcs" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="sim" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="tim" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="fct" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
-	<xs:element name="stn" type="projectElement" minOccurs="0" maxOccurs="unbounded" />
+        <xs:element name="input" minOccurs="1" maxOccurs="1">
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="mesh" type="MeshFileType" minOccurs="0" maxOccurs="unbounded" />
+              <xs:element name="geometry" type="FileType" minOccurs="0" maxOccurs="unbounded" />
+              <xs:element name="stations" type="FileType" minOccurs="0" maxOccurs="unbounded" />
+            </xs:sequence>
+          </xs:complexType>
+        </xs:element>
+
+        <xs:element name="processes" minOccurs="0" maxOccurs="1">
```

Comment:
no, every child of the root element (except input files) need to have minOccurs=0 so you can build a project from input files.
once you want to start simulation you will need 1 but that needs to be checked in config tree.